src/cryptonote_core/miner.h

```diff
@@ -119,8 +147,37 @@ namespace cryptonote
     bool m_do_print_hashrate;
     bool m_do_mining;
 
-  };
-}
-
+    // background mining stuffs ..
 
+    //static const int CPU_DATA_OVERALL_TIMES_COUNT = 4;
+    //static const int CPU_DATA_OVERALL_USER_INDEX = 0;
+    //static const int CPU_DATA_OVERALL_NICE_INDEX = 1;
+    //static const int CPU_DATA_OVERALL_SYSTEM_INDEX = 2;
+    //static const int CPU_DATA_OVERALL_IDLE_INDEX = 3;
+    //const int CPU_DATA_PROCESS_TIMES_COUNT;
+    //struct cpu_data
+    //{
+    //  uint64_t update_time; // set to 0 to update
+    //  uint64_t overall_cpu_times[CPU_DATA_OVERALL_TIMES_COUNT];
+    //  uint64_t process_cpu_time;
+    //};
 
+    bool background_worker_thread();
+    volatile bool m_is_background_mining_enabled;
```

Comment:
Just thinking future-wise, since the function is there, what if someone calls set_is_background_mining_enabled(false) after the threads are created/started? Ideally the background logic would be the only thing to stop and mining would continue full speed.